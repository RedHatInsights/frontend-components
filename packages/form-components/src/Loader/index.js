import React from 'react';
import { Skeleton } from '@redhat-cloud-services/frontend-components/components/Skeleton';
import PropTypes from 'prop-types';

// eslint-disable-next-line no-unused-vars
const Loader = ({ name, size, FieldProvider, validate, FormSpyProvider, formOptions, ...rest }) => (
    <div { ...rest } >
        <Skeleton size={ size }></Skeleton>
    </div>
);

Loader.propTypes = {
    FieldProvider: PropTypes.any,
    formOptions: PropTypes.any,
    FormSpyProvider: PropTypes.any,
    validate: PropTypes.any,
    name: PropTypes.string,
    size: PropTypes.string
};

Loader.defaultProps = {
    name: '',
    size: 'md'
};

export default Loader;
