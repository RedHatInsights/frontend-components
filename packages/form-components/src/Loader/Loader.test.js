import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loader from './';

describe('Loader component tests', () => {

    it('should render correctly', () => {
        const wrapper = mount(
            <Loader size="sm"></Loader>
        );
        expect(toJson(wrapper.find('Loader'))).toMatchSnapshot();
    });
});
