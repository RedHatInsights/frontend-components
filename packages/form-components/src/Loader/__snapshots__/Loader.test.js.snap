// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component tests should render correctly 1`] = `
<Loader
  name=""
  size="sm"
>
  <div>
    <Skeleton
      isDark={false}
      size="sm"
    >
      <div
        className="ins-c-skeleton ins-c-skeleton__sm"
      >
        Â 
      </div>
    </Skeleton>
  </div>
</Loader>
`;
